operational/snapshots/no_recent_snapshots.pt

```diff
@@ -0,0 +1,262 @@
+    state "operational"
+  end
+  tags none($param_exclude_tags)
+end
+
+datasource "ds_instances" do
+  iterate @instances
+  field "resource_uid", val(iter_item,  "resource_uid")
+  field "name", val(iter_item, "name")
+  field "href", href(iter_item)
+  field "cloud_href", jmes_path(iter_item, "links[?rel=='cloud'].href | [0]")
+end
+
+resources "volume_attachments", type: "rs_cm.volume_attachments" do
+  iterate @instances
+  cloud_href jmes_path(iter_item, "links[?rel=='cloud'].href | [0]")
+  filter do
+    instance_href href(iter_item)
+  end
+end
+
+datasource "ds_volume_attachments" do
+  iterate @volume_attachments
+  field "href", href(iter_item)
+  field "device", val(iter_item, "device")
+  field "volume_href", jmes_path(iter_item, "links[?rel=='volume'].href | [0]")
+  field "instance_href", jmes_path(iter_item, "links[?rel=='instance'].href | [0]")
+end
+
+resources "volumes", type: "rs_cm.volumes" do
+  iterate @clouds  
+  cloud_href href(iter_item)
+  tags none($param_exclude_tags)
+end
+
+datasource "ds_volumes" do
+  iterate(@volumes)
+  field "resource_uid", val(iter_item,  "resource_uid")
+  field "name", val(iter_item, "name")
+  field "href", href(iter_item)
+end
+
+resources "volume_snapshots", type: "rs_cm.volume_snapshots" do
+  iterate @clouds  
+  cloud_href href(iter_item)
+end
+
+datasource "ds_volume_snapshots" do
+  iterate(@volume_snapshots)
+  field "resource_uid", val(iter_item,  "resource_uid")
+  field "name", val(iter_item, "name")
+  field "href", href(iter_item)
+  field "parent_volume_href", jmes_path(iter_item, "links[?rel=='parent_volume'].href | [0]")
+  field "updated_at", val(iter_item, "updated_at")
+end
+
+datasource "ds_volumes_volume_snapshots" do
+  run_script $js_volumes_volume_snapshots, $ds_volume_attachments, $ds_volumes, $ds_volume_snapshots, $ds_instances, $param_include_root_device
+end
+
+script "js_volumes_volume_snapshots", type: "javascript" do
+  parameters "ds_volume_attachments", "ds_volumes", "ds_volume_snapshots", "ds_instances", "include_root_device"
+  result "volumes_volume_snapshots"
+  code <<-EOS
+  var volumes_volume_snapshots = []
+  var volumes = {}
+  for (var index = 0; index < ds_volumes.length; index++) {
+    var volume = ds_volumes[index];
+    volumes[volume.href] = volume;
+  }
+  var snapshots = {}
+  for (var index = 0; index < ds_volume_snapshots.length; index++) {
+    var snapshot = ds_volume_snapshots[index]
+    if (snapshots[snapshot.parent_volume_href]) {
+      snapshots[snapshot.parent_volume_href].push(snapshot)
+    }else {
+      snapshots[snapshot.parent_volume_href] = [snapshot]
+    }
+  }
+
+  var instances = {}
+  for (var index = 0; index < ds_instances.length; index++) {
+    var instance = ds_instances[index]
+    instances[instance.href] = instance
+  }
+
+  for ( i = 0; i < ds_volume_attachments.length; i++) {
+    device = ds_volume_attachments[i].device
+    if ( device == "/dev/sda1" && include_root_device == 0 ) { continue; }

```

Comment:
I don't think root disk always shows up as device = /dev/sda1
For example, this is from an AzureRM instance in account 30601 and device is osDisk:
[
    {
        "actions": [],
        "created_at": "2018/06/19 14:26:39 +0000",
        "device": "osDisk",
        "device_id": "osDisk",
        "links": [
            {
                "href": "/api/clouds/3523/volume_attachments/1SHHL2FPFK4UB",
                "rel": "self"
            },
            {
                "href": "/api/clouds/3523",
                "rel": "cloud"
            },
            {
                "href": "/api/clouds/3523/volumes/A6MAL89EE7EKP",
                "rel": "volume"
            },
            {
                "href": "/api/clouds/3523/instances/E4UJQ3D3OEOPM",
                "rel": "instance"
            }
        ],
        "resource_uid": "joshdelete3es84954c10fbc@vhds@os-App-113485800313016dc797-rs.vhd:9cc6e4d5-4ff8-4d5b-a994-2c498ac507de",
        "state": "attached",
        "updated_at": "2018/06/19 14:26:39 +0000"
    }
]